api/app/Models/AwardExperience.php

```diff
@@ -25,21 +27,86 @@ class AwardExperience extends Experience
     use SoftDeletes;
 
     /**
-     * The attributes that should be cast.
+     * The table associated with the model.
      *
-     * @var array
+     * @var string
      */
-    protected $casts = [
-        'awarded_date' => 'date',
+    protected $table = 'experiences';
+
+    /**
+     * Default values for attributes
+     *
+     * @var array an array with attribute as key and default as value
+     */
+    protected $attributes = [
+        'experience_type' => AwardExperience::class,
     ];
 
     public function getTitle(): string
     {
         return $this->title;
     }
 
-    public function getExperienceType(): string
+    /**
+     * The "booted" method of the model.
+     */
+    protected static function booted(): void
+    {
+        static::addGlobalScope(new MatchExperienceType);
+    }
+
+    /**
+     * Interact with the experience's title
+     */
+    protected function title(): Attribute
+    {
+        return Attribute::make(
+            get: fn (mixed $value, array $attributes) => $this::getJsonPropertyString($attributes, 'title'),
+            set: fn (mixed $value, array $attributes) => $this::setJsonPropertyString($value, $attributes, 'title')
+        );
```

Comment:
This pattern is repeated so much I might even abstract it a bit more 😆 Not that this is a necessary change or anything.

```
function makeJsonStringAttribute($attribute_name)
{
    return Attribute::make(
        get: fn (mixed $value, array $attributes) => $this::getJsonPropertyString($attributes, $attribute_name),
        set: fn (mixed $value, array $attributes) => $this::setJsonPropertyString($value, $attributes, $attribute_name)
    );
}
```

Reply:
I gave it a try.  It's a little drier, though (IMO) a little harder to follow.  Let me know which way you prefer.
[make attributes drier](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9899/commits/5b07433e3d651b42780ac782d12e6c33fb681a22)